extensions/wikia/Email/Controller/DiscussionController.class.php

```diff
@@ -356,3 +356,47 @@ protected static function getEmailSpecificFormFields() {
         return array_merge_recursive( parent::getEmailSpecificFormFields(), $formFields );
     }
 }
+
+class DiscussionArticleCommentController extends DiscussionController {
+
+	private $contentType;
+	private $articleTitle;
+
+	const ARTICLE_COMMENT_REPLY = 'article-comment-reply';
+	const ARTICLE_COMMENT_AT_MENTION = 'article-comment-at-mention';
+	const ARTICLE_COMMENT_REPLY_AT_MENTION = 'article-comment-reply-at-mention';
+
+    public function initEmail() {
+		$this->articleTitle = $this->request->getVal( 'threadTitle' );
```

Comment:
Is it the title of a thread or an article? Can we unify naming in request and class param?

Reply:
It is article title. It's the only place where I can put the article title without expanding the structure. Thread title is also used in other requests, but for a name of a thread. These structures for sending data to this controller are quite limited and should remain generic as possible. I agree it's kind of confusing. I could at least add a comment.

Reply:
This is how it works right now. `notification` service sends request that is handled by this controller with [query params](https://github.com/Wikia/pandora/blob/master/service/notification/src/main/java/com/wikia/services/notification/service/dispatcher/email/MediaWikiEmailApi.java). One of this param is threadTitle. I could've added another field but IMO there's no need to add another piece just for the sake of name.